In [2]:
import pandas as pd

In [3]:
df=pd.read_csv(r'C:\Users\QC#\AppData\Local\Packages\5319275A.WhatsAppDesktop_cv1g1gvanyjgm\LocalState\sessions\DEA49B93F89B7BF917116DD6B20A47509C5F70B1\transfers\2026-08\ecommerce_sales_data.csv')

In [4]:
df

,Year,Month,Category,Region,Units_Sold,Revenue,Profit,Customer_Rating,Ad_Spend
0,2020,April,Home Decor,North,485,94360.58,25442.39,1.5,174.51
1,2021,October,Fashion,South,95,14081.40,3457.23,3.5,3975.18
2,2022,February,Groceries,East,182,13700.50,3985.61,3.4,1929.31
3,2020,April,Fashion,South,419,81856.23,20726.68,2.8,3706.92
4,2020,September,Beauty Products,North,469,51810.41,5624.12,2.9,2362.07
...,...,...,...,...,...,...,...,...,...
995,2023,January,Fashion,West,38,5606.14,1577.25,3.2,4788.84
996,2024,October,Groceries,East,422,13513.47,2667.70,3.1,495.29
997,2021,January,Electronics,North,85,14304.92,2047.38,2.1,370.55
998,2024,May,Groceries,South,434,66941.72,14621.33,2.5,2944.75


In [5]:
df.isna().sum()

Year               0
Month              0
Category           0
Region             0
Units_Sold         0
Revenue            0
Profit             0
Customer_Rating    0
Ad_Spend           0
dtype: int64

In [6]:
df.duplicated().sum()

np.int64(0)

In [7]:
df['Date']=pd.to_datetime(df['Year'].astype(str) + '-' +
                          df['Month'], format= '%Y-%B')

In [8]:
df.sort_values('Date', 
               inplace=True)

In [9]:
df

,Year,Month,Category,Region,Units_Sold,Revenue,Profit,Customer_Rating,Ad_Spend,Date
468,2020,January,Home Decor,North,138,26033.04,5069.60,3.5,4342.90,2020-01-01
168,2020,January,Home Decor,East,301,29655.09,8341.07,3.7,173.97,2020-01-01
760,2020,January,Electronics,West,438,30470.06,4476.40,2.7,4132.49,2020-01-01
559,2020,January,Electronics,South,65,3199.28,381.00,1.3,4025.14,2020-01-01
497,2020,January,Beauty Products,East,80,4633.99,766.94,1.1,3403.18,2020-01-01
...,...,...,...,...,...,...,...,...,...,...
21,2024,December,Electronics,South,346,41810.49,9999.55,3.9,404.29,2024-12-01
254,2024,December,Beauty Products,South,298,19371.61,4780.94,1.4,908.76,2024-12-01
897,2024,December,Electronics,East,428,80545.84,20172.34,2.7,4712.18,2024-12-01
605,2024,December,Home Decor,West,226,13724.82,1617.18,4.1,4419.21,2024-12-01


In [10]:
# total revenue
df['Revenue'].sum()

np.float64(27123544.52)

In [11]:
#top customer by revenue
df.groupby(['Category','Region'])['Revenue'].sum().sort_values(ascending=False)

Category         Region
Home Decor       North     1527233.10
Beauty Products  North     1387055.05
Groceries        East      1372648.57
Electronics      North     1323411.45
Home Decor       East      1314584.94
Groceries        South     1290353.96
                 North     1257999.12
Beauty Products  East      1181545.32
                 South     1154764.87
Fashion          South     1145363.06
                 West      1131966.74
Electronics      West      1118084.76
Fashion          North     1116154.87
Home Decor       South     1113012.48
Electronics      South     1103348.21
                 East      1096083.84
Books            North     1067976.38
Home Decor       West      1066688.65
Books            East      1010647.30
Beauty Products  West       975700.45
Groceries        West       868066.25
Fashion          East       862704.40
Books            West       855822.06
                 South      782328.69
Name: Revenue, dtype: float64

In [12]:
#monthly Revenue tend
monthly_revenue=df.groupby(['Month'])['Revenue'].sum().sort_values(ascending=False)

In [13]:
monthly_revenue

Month
June         2704548.43
October      2640283.17
February     2556542.09
May          2539650.55
January      2271989.21
November     2265159.35
July         2100090.77
December     2096213.46
September    2053765.68
August       2046098.80
March        1965211.31
April        1883991.70
Name: Revenue, dtype: float64

In [14]:
#Yearly Revenue
yearly_Revenue=df.groupby(['Year'])['Revenue'].sum().sort_values(ascending=False)

In [15]:
yearly_Revenue

Year
2022    6734155.76
2021    5377030.28
2024    5283868.18
2020    4910693.47
2023    4817796.83
Name: Revenue, dtype: float64

In [16]:
#which category generate most revenue
df.groupby('Category')['Revenue'].sum().sort_values(ascending=False)

Category
Home Decor         5021519.17
Groceries          4789067.90
Beauty Products    4699065.69
Electronics        4640928.26
Fashion            4256189.07
Books              3716774.43
Name: Revenue, dtype: float64

In [17]:
#Most profitable category
df.groupby('Category')['Profit'].sum().sort_values(ascending=False)

Category
Groceries          1012596.35
Home Decor         1010554.50
Electronics         929368.70
Beauty Products     919309.22
Fashion             879845.52
Books               743058.37
Name: Profit, dtype: float64

In [18]:
#profit margin
df['profit_margin']=df['Profit']/df['Revenue']*100

In [19]:
#Profit margin by category
df.groupby('Category')['profit_margin'].mean().sort_values(ascending=False).map("{:.1f}%".format)

Category
Groceries          21.1%
Fashion            20.9%
Home Decor         20.3%
Electronics        19.5%
Beauty Products    19.5%
Books              19.2%
Name: profit_margin, dtype: str

In [20]:
df=df.reset_index(drop=True)
df.head(10)

,Year,Month,Category,Region,Units_Sold,Revenue,Profit,Customer_Rating,Ad_Spend,Date,profit_margin
0,2020,January,Home Decor,North,138,26033.04,5069.60,3.5,4342.90,2020-01-01,19.473715
1,2020,January,Home Decor,East,301,29655.09,8341.07,3.7,173.97,2020-01-01,28.126942
2,2020,January,Electronics,West,438,30470.06,4476.40,2.7,4132.49,2020-01-01,14.691143
3,2020,January,Electronics,South,65,3199.28,381.00,1.3,4025.14,2020-01-01,11.908930
4,2020,January,Beauty Products,East,80,4633.99,766.94,1.1,3403.18,2020-01-01,16.550316
5,2020,January,Books,East,329,50397.26,12180.98,2.3,3399.33,2020-01-01,24.169925
6,2020,January,Books,West,269,13987.85,2167.47,4.7,2076.47,2020-01-01,15.495376
7,2020,January,Fashion,West,361,29919.66,3814.91,3.1,109.77,2020-01-01,12.750513
8,2020,January,Home Decor,West,234,17168.96,4096.52,2.9,3718.93,2020-01-01,23.860036
9,2020,January,Groceries,South,107,3058.58,519.64,4.8,2749.84,2020-01-01,16.989583


In [21]:
#Best region by revenue
df.groupby('Region')['Revenue'].sum().sort_values(ascending=False)

Region
North    7679829.97
East     6838214.37
South    6589171.27
West     6016328.91
Name: Revenue, dtype: float64

In [22]:
#Best region by profit
df.groupby('Region')['Profit'].sum().sort_values(ascending=False)

Region
North    1544081.78
East     1357339.37
South    1341084.34
West     1252227.17
Name: Profit, dtype: float64

In [23]:
#Which category has high ad spend & But low profit
df.groupby('Category')[['Profit' , 'Ad_Spend']].sum().sort_values(by='Profit',ascending=False)

,Profit,Ad_Spend
Category,,
Groceries,1012596.35,423612.73
Home Decor,1010554.50,454823.34
Electronics,929368.70,475469.54
Beauty Products,919309.22,408330.75
Fashion,879845.52,438873.67
Books,743058.37,448444.58


In [24]:
#AD_spend increase Revenue
df[['Ad_Spend','Revenue']].corr()

,Ad_Spend,Revenue
Ad_Spend,1.000000,0.015863
Revenue,0.015863,1.000000


In [25]:
#Create Roas column
#Roas=For that how much return ads generate
df['Roas']=df['Revenue']/df['Ad_Spend']


In [26]:
df.head()

,Year,Month,Category,Region,Units_Sold,Revenue,Profit,Customer_Rating,Ad_Spend,Date,profit_margin,Roas
0,2020,January,Home Decor,North,138,26033.04,5069.60,3.5,4342.90,2020-01-01,19.473715,5.994391
1,2020,January,Home Decor,East,301,29655.09,8341.07,3.7,173.97,2020-01-01,28.126942,170.460942
2,2020,January,Electronics,West,438,30470.06,4476.40,2.7,4132.49,2020-01-01,14.691143,7.373293
3,2020,January,Electronics,South,65,3199.28,381.00,1.3,4025.14,2020-01-01,11.908930,0.794825
4,2020,January,Beauty Products,East,80,4633.99,766.94,1.1,3403.18,2020-01-01,16.550316,1.361665


In [27]:
df.groupby('Category')['Roas'].mean().sort_values(ascending=False)

Category
Home Decor         24.169183
Beauty Products    21.075111
Electronics        19.544035
Groceries          18.896417
Fashion            17.757200
Books              13.351887
Name: Roas, dtype: float64

In [29]:
#which category has  avg Rating.
df.groupby('Category')['Customer_Rating'].mean().sort_values(ascending=False)

Category
Home Decor         3.137209
Fashion            3.104192
Beauty Products    3.029167
Groceries          3.016981
Books              3.008974
Electronics        2.944944
Name: Customer_Rating, dtype: float64

In [30]:
#Rating impact on Revenue
df[['Customer_Rating', 'Revenue']].corr()

,Customer_Rating,Revenue
Customer_Rating,1.000000,0.036835
Revenue,0.036835,1.000000


# Insights
1:
~Highest revenue and profit is from home_decore
~and avg rating of home decore is also high
~This is our Stable And star category
2:
~while home decore make the most money but Grocieries is actually the most efficient 21.1% profit_margin
we want to Scale the bussiness in quickly,investment more in Groceries migh Yield better per Unit return then Other Categories
3:
~The north region is our power house by Total Revenue and Profit
4:
~Correlation is week between b/w Ad_spend and revenue.Our currently ads are not efficient driving Sales.
5:
~June is Strongest month in Revenue.2022 is best year



